File: ui/src/main/java/org/wcl/mfn/ui/model/SuggestedYearContract.java
package org.wcl.mfn.ui.model;

import java.util.List;
import java.util.stream.IntStream;

public record SuggestedYearContract(int years, int totalSalary, int totalRemuneration,
                                    List<Integer> yearlySalaries, int bonusPerYear, List<Integer> totalsPerYear) {
    public List<Integer> listedYears() {
        return IntStream.rangeClosed(1,years).boxed().toList();
    }
}

File: ui/src/main/kotlin/org/wcl/mfn/ui/model/tools/contractcalculator/SuggestedYearContract.kt
package org.wcl.mfn.ui.model.tools.contractcalculator

import java.util.stream.IntStream

data class SuggestedYearContract(
    val years: Int, val totalSalary: Int, val totalRemuneration: Int,
    val yearlySalaries: List<Int>, val bonusPerYear: Int, val totalsPerYear: List<Int>
) {
    fun listedYears(): List<Int> {
        return IntStream.rangeClosed(1, years).boxed().toList()
    }
}
